embedding_pdf_documents/embedding_ray.py

```diff
@@ -0,0 +1,96 @@
+import binascii
+import io
+from typing import List
+
+import pypdf
+import ray
+from pypdf import PdfReader
+
+ray.init(
+    runtime_env={"pip": ["langchain", "pypdf", "sentence_transformers", "transformers"]}
+)
+
+ds = ray.data.read_binary_files("s3://arxiv-docs/")
+
+
+def convert_to_text(pdf_bytes: bytes):
+    pdf_bytes_io = io.BytesIO(pdf_bytes)
+
+    try:
+        pdf_doc = PdfReader(pdf_bytes_io)
+    except pypdf.errors.PdfStreamError:
+        # Skip pdfs that are not readable.
+        # We still have over 30,000 pages after skipping these.
+        return []
+
+    text = []
+    for page in pdf_doc.pages:
+        try:
+            text.append(page.extract_text())
+        except binascii.Error:
+            # Skip all pages that are not parseable due to malformed characters.
+            print("parsing failed")
+    return text
+
+
+ds = ds.flat_map(convert_to_text)
+
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+
+def split_text(page_text: str):
+    text_splitter = RecursiveCharacterTextSplitter(
+        chunk_size=1000, chunk_overlap=100, length_function=len
```

Comment:
Explain where you got these numbers from. E.g. "We felt that the answer we would be looking for would be around 200 words, or around 1000 characters."